internal/command/format/diff.go

```diff
@@ -489,48 +486,62 @@ func (p *blockBodyDiffPrinter) writeNestedAttrDiff(
 		// commonLen is number of elements that exist in both lists, which
 		// will be presented as updates (~). Any additional items in one
 		// of the lists will be presented as either creates (+) or deletes (-)
-		// depending on which list they belong to.
+		// depending on which list they belong to. maxLen is the number of
+		// elements in that longer list.
 		var commonLen int
+		var maxLen int
 		// unchanged is the number of unchanged elements
 		var unchanged int
 
 		switch {
 		case len(oldItems) < len(newItems):
 			commonLen = len(oldItems)
+			maxLen = len(newItems)
 		default:
 			commonLen = len(newItems)
+			maxLen = len(oldItems)
 		}
-		for i := 0; i < commonLen; i++ {
+		for i := 0; i < maxLen; i++ {
 			path := append(path, cty.IndexStep{Key: cty.NumberIntVal(int64(i))})
-			oldItem := oldItems[i]
-			newItem := newItems[i]
-			if oldItem.RawEquals(newItem) {
+
+			var action plans.Action
+			var oldItem, newItem cty.Value
+			switch {
+			case i < commonLen:
+				oldItem = oldItems[i]
+				newItem = newItems[i]
+				if oldItem.RawEquals(newItem) {
+					action = plans.NoOp
+					unchanged++
+				} else {
+					action = plans.Update
+				}
+			case i < len(oldItems):
+				oldItem = oldItems[i]
+				newItem = cty.NullVal(oldItem.Type())
+				action = plans.Delete
+			case i < len(newItems):
+				newItem = newItems[i]
+				oldItem = cty.NullVal(newItem.Type())
+				action = plans.Create
+			default:
 				action = plans.NoOp
-				unchanged++
 			}
+
 			if action != plans.NoOp {
-				p.writeAttrsDiff(objS.Attributes, oldItem, newItem, indent+6, path, result)
-				p.writeSkippedAttr(result.skippedAttributes, indent+8)
+				p.buf.WriteString(strings.Repeat(" ", indent+4))
+				p.writeActionSymbol(action)
+				p.buf.WriteString("{")
+				p.writeAttrsDiff(objS.Attributes, oldItem, newItem, indent+8, path, result)
+				if action == plans.Update {
+					p.writeSkippedAttr(result.skippedAttributes, indent+10)
```

Comment:
We're reusing the same `blockBodyDiffResult ` value for multiple calls here, which means that the `skippedAttributes` field will continue to increment with every element. Screenshot example, where all of these values have one skipped attribute:

<img width="465" alt="image" src="https://user-images.githubusercontent.com/68917/139730503-ebecfd38-aead-48ac-8164-9de5dfa2ccab.png">

Reply:
Good catch, fixed and extended the tests to cover this.